server/service/integration_mdm_dep_test.go

```diff
@@ -0,0 +1,1083 @@
+		//default:
+		//	fmt.Println(">>>> device received command: ", cmd.CommandUUID, cmd.Command.RequestType)
+		//}
+		cmds = append(cmds, cmd)
+		cmd, err = mdmDevice.Acknowledge(cmd.CommandUUID)
+		require.NoError(t, err)
+	}
+
+	// expected commands: install fleetd, install bootstrap, install profiles
+	// (custom one and fleetd configuration) (not expected: account
+	// configuration, since enrollment_reference not set)
+	require.Len(t, cmds, 4)
+	var installProfileCount, installEnterpriseCount, otherCount int
+	var profileCustomSeen, profileFleetdSeen bool
+	for _, cmd := range cmds {
+		switch cmd.Command.RequestType {
+		case "InstallProfile":
+			installProfileCount++
+			if strings.Contains(string(cmd.Command.InstallProfile.Payload), fmt.Sprintf("<string>%s</string>", customProfileIdent)) {
+				profileCustomSeen = true
+			} else if strings.Contains(string(cmd.Command.InstallProfile.Payload), fmt.Sprintf("<string>%s</string>", mobileconfig.FleetdConfigPayloadIdentifier)) {
+				profileFleetdSeen = true
+			}
+
+		case "InstallEnterpriseApplication":
+			installEnterpriseCount++
+		default:
+			otherCount++
+		}
+	}
+	require.Equal(t, 2, installProfileCount)
+	require.Equal(t, 2, installEnterpriseCount)
+	require.Equal(t, 0, otherCount)
+	require.True(t, profileCustomSeen)
+	require.True(t, profileFleetdSeen)
+
+	if enableReleaseManually {
+		// get the worker's pending job from the future, there should not be any
+		// because it needs to be released manually
+		pending, err := s.ds.GetQueuedJobs(ctx, 1, time.Now().UTC().Add(time.Minute))
+		require.NoError(t, err)
+		require.Empty(t, pending)
+	} else {
+		// get the worker's pending job from the future, there should be a DEP
+		// release device task
+		pending, err := s.ds.GetQueuedJobs(ctx, 1, time.Now().UTC().Add(time.Minute))
+		require.NoError(t, err)
+		require.Len(t, pending, 1)
+		releaseJob := pending[0]
+		require.Equal(t, 0, releaseJob.Retries)
+		require.Contains(t, string(*releaseJob.Args), worker.AppleMDMPostDEPReleaseDeviceTask)
+
+		// update the job so that it can run immediately
+		releaseJob.NotBefore = time.Now().UTC().Add(-time.Minute)
+		_, err = s.ds.UpdateJob(ctx, releaseJob.ID, releaseJob)
+		require.NoError(t, err)
+
+		// run the worker to process the DEP release
+		s.runWorker()
+
+		// make the device process the commands, it should receive the
+		// DeviceConfigured one.
+		cmds = cmds[:0]
+		cmd, err = mdmDevice.Idle()
+		require.NoError(t, err)
+		for cmd != nil {
+			cmds = append(cmds, cmd)
+			cmd, err = mdmDevice.Acknowledge(cmd.CommandUUID)
+			require.NoError(t, err)
+		}
+
+		require.Len(t, cmds, 1)
+		var deviceConfiguredCount int
+		for _, cmd := range cmds {
+			switch cmd.Command.RequestType {
+			case "DeviceConfigured":
+				deviceConfiguredCount++
+			default:
+				otherCount++
+			}
+		}
+		require.Equal(t, 1, deviceConfiguredCount)
+		require.Equal(t, 0, otherCount)
+	}
+}
+
+func (s *integrationMDMTestSuite) TestDEPProfileAssignment() {
```

Comment:
This test was not modified, just moved here from the `integration_mdm_test.go` as it is DEP-specific.